tm2/pkg/bft/state/store.go

```diff
@@ -172,6 +180,27 @@ func saveABCIResponses(db dbm.DB, height int64, abciResponses *ABCIResponses) {
 	db.SetSync(calcABCIResponsesKey(height), abciResponses.Bytes())
 }
 
+// LoadTxResult loads the tx result associated with the given
+// tx hash from the database, if any
+func LoadTxResult(db dbm.DB, txHash []byte) (*types.TxResult, error) {
+	buf := db.Get(CalcTxResultKey(txHash))
+	if buf == nil {
+		return nil, NoTxResultForHashError{txHash}
+	}
+
+	txResult := new(types.TxResult)
+	if err := amino.Unmarshal(buf, txResult); err != nil {
+		return nil, fmt.Errorf("%w, %w", errTxResultCorrupted, err)
+	}
+
+	return txResult, nil
+}
+
+// saveTxResult persists the transaction result to the database
+func saveTxResult(db dbm.DB, txResult *types.TxResult) {
+	db.SetSync(CalcTxResultKey(txResult.Tx.Hash()), txResult.Bytes())
```

Comment:
maybe SetSync is unnecessary... to do a synchronous set per tx sounds potentially pretty expensive. 
usually there is a sync after a block has been processed so it may not be necessary at all.

Reply:
I've looked into it, we do a CommitSave just shorty after.

I've swapped `SetSync` to `Set`, for the tx results:

c627995